resources/external_control.urscript

```diff
@@ -77,6 +84,48 @@ thread_move = 0
 thread_trajectory = 0
 thread_script_commands = 0
 
+###
+# @brief Function to verify whether the specified target can be reached within the defined time frame while staying within
+# the robot's speed limits
+#
+# @param target array is the joint target to reach
+# @param time float is the time to reach the target
+#
+# @returns bool true if the target is reachable within the robot's speed limits, false otherwise
+###
+def targetWithinLimits(step_start, step_end, time):
+  local idx = 0
+  while idx < 6:
+    local velocity = norm(step_end[idx] - step_start[idx]) / time
+    if velocity > JOINT_IGNORE_SPEED:
+      local str = str_cat(str_cat("Velocity ", velocity), str_cat(str_cat(" required to reach the received target ", step_end), str_cat(str_cat(" within ", time), " seconds is exceeding the joint velocity limits. Ignoring commands until a valid command is received.")))
+      if last_violation_popup_counter == 0:
+        textmsg(str)
+        popup(str, blocking=False)
+      elif last_violation_popup_counter * get_steptime() > 5.0:
+        # We want to have a log printed regularly. We are receiving motion commands that are not
+        # feasible. The user should have a chance to know about this.
+        textmsg(str)
+        last_violation_popup_counter = 0
+      end
+      last_violation_popup_counter = last_violation_popup_counter + 1
```

Comment:
```suggestion
      if last_violation_popup_counter == 0:
        # Keep making a popup as long as the targets are too much off
        local str = str_cat(str_cat("Velocity ", velocity), str_cat(str_cat(" required to reach the received target ", step_end), str_cat(str_cat(" within ", time), " seconds is exceeding the joint velocity limits. Ignoring commands until a valid command is received.")))
        textmsg(str)
        popup(str, error=True, blocking=False)
      end
      # Report the error every 5 sec
      last_violation_popup_counter = last_violation_popup_counter + 1
      if last_violation_popup_counter * get_steptime() >= 5.0:
        last_violation_popup_counter = 0
      end
```
I think it is fine it is not blocking, as the alterative is more to terminating the program instead. Though let's ensure that the popup keeps coming if a continues stream of invalid targets keeps coming